File: src/daemon/paths.ts
import { homedir } from "node:os";
import { join } from "node:path";
import { createHash } from "node:crypto";

export const RUNTIME_DIR = join(homedir(), ".agent-database-cli");
export const SOCKET_PATH = resolveSocketPath(process.platform, homedir());
export const PID_PATH = join(RUNTIME_DIR, "agent-database-cli.pid");

export function resolveSocketPath(platform: NodeJS.Platform, homeDir: string): string {
  if (platform === "win32") {
    // Windows IPC 必须使用 named pipe，不能使用 Unix socket 文件路径。
    const suffix = createHash("sha1").update(homeDir).digest("hex").slice(0, 12);
    return `\\\\.\\pipe\\agent-database-cli-${suffix}`;
  }
  return join(homeDir, ".agent-database-cli", "agent-database-cli.sock");
}

export function isWindowsNamedPipe(path: string): boolean {
  return path.startsWith("\\\\.\\pipe\\");
}

File: rust/src/daemon/paths.rs
use anyhow::{Context, Result};
use std::path::PathBuf;

pub fn runtime_dir() -> Result<PathBuf> {
    Ok(dirs::home_dir()
        .context("无法解析用户主目录")?
        .join(".agent-database-cli"))
}

pub fn pid_path() -> Result<PathBuf> {
    Ok(runtime_dir()?.join("agent-database-cli.pid"))
}

#[cfg(unix)]
pub fn socket_path() -> Result<PathBuf> {
    Ok(runtime_dir()?.join("agent-database-cli.sock"))
}

#[cfg(windows)]
pub fn socket_path_string() -> Result<String> {
    use sha1::{Digest, Sha1};
    let home = dirs::home_dir().context("无法解析用户主目录")?;
    let mut hasher = Sha1::new();
    hasher.update(home.to_string_lossy().as_bytes());
    let digest = format!("{:x}", hasher.finalize());
    Ok(format!(r"\\.\pipe\agent-database-cli-{}", &digest[..12]))
}
